### PR TITLE
Phase N: hardening stack (file splits + 3 new warnings + docs)

### PR DESCRIPTION
**Pull request by Ethan-IS**
Single-PR consolidation of Phase N — post-Phase-M hardening track stacked on #25 (`codex/refactor-hierarchy-validator-prune-tca-example`). Ten commits across four sub-phases (N-1, N-3, N-4, N-5; N-2 was made obsolete by earlier Sendable decisions — see Notes). No behavior change for existing users: N-1 is pure file reorganization, N-3/N-4 add warning-severity diagnostics only, N-5 is docs + one edge-case test.

## 구성

| Sub-phase | 내용 | 커밋 수 |
|---|---|---|
| N-1 | 4개 거대 파일 분할 (1,259+904+744+722줄 → 4 entry + 9 new siblings) | 5 (원본 4개 + 리뷰 수정 1개) |
| N-3 | \`sub.child-overrides-missing\` 경고 + review 수정 | 2 |
| N-4 | \`provide.lazy-aliased\` / \`provide.provider-aliased\` 경고 | 1 |
| N-5 | README/README.ko/MIGRATION/CHANGELOG/ROADMAP + bindings 다중 remap 스냅샷 + review 수정 | 2 |

## 설계 결정

- **스냅샷 불변이 N-1 성공 기준**: 4개 파일을 쪼개는 동안 모든 매크로 확장 스냅샷이 byte-identical로 유지됨.
- **경고 severity, error 아님**: N-3/N-4 모두 기존 코드를 깨지 않도록 warning으로 한정. 탐지는 best-effort, same-file만. cross-module 커버리지는 build-support validator의 후속 PR 과제.
- **문서 명확화**: \`@DIContainer(root:)\`는 CLI 그래프 렌더링 전용 플래그임을 README 매트릭스로 문서화. \`validateDAG\`와 독립.
- **\`bindings:\` 문서화**: 기존 \`with:\` 서브셋 shorthand 외에 rename-aware \`bindings:\` 예시를 README에 first-class로 소개.

### N-1 파일 분할 (9 new siblings)

| 기존 (줄수) | 분할 결과 (entry + new siblings) |
|---|---|
| \`DIContainerCodeGenerator.swift\` (1,259) | entry (656) + \`DIContainerOverridesGenerator.swift\` + \`DIContainerWithOverridesGenerator.swift\` + \`DIContainerSubContainerGenerator.swift\` |
| \`DIContainerValidator.swift\` (904) | entry (418) + \`DIProvideValidationDiagnostics.swift\` + \`DIContainerValidatorTypeChecks.swift\` |
| \`ValidationCoordinator.swift\` (744) | entry (485) + \`ValidationCoordinator+Caching.swift\` + \`ValidationCoordinator+Locking.swift\` |
| \`SwiftUIMacros.swift\` (722) | entry (216) + \`DIEnvironmentBridgeMacro.swift\` + \`DIFeatureRootMacro.swift\` |

### N-3/N-4 신규 진단 인프라

- \`DISubContainerChildOverridesCheck.swift\` (NEW) — child scope 멤버십 검사
- \`DILazyProviderAliasCheck.swift\` (NEW) — 소스 파일의 \`typealias\` 수집 + closure 파라미터 조회
- \`Diagnostics.swift\` — 3개 신규 코드 (\`sub.child-overrides-missing\`, \`provide.lazy-aliased\`, \`provide.provider-aliased\`)
- \`DIContainerValidator.swift\` — \`subsWithErrors\` 게이트로 중복 경고 방지, alias 검사 hook

### N-5 문서/테스트

- \`README.md\` / \`README.ko.md\` — \`root\` vs \`validateDAG\` 매트릭스, \`bindings:\` 예시, 진단 테이블 확장 (N-3, bindings 관련 코드 4개 추가)
- \`MIGRATION.md\` — Phase N 엔트리 (file relocations + 2개 warnings + \`root\` 명확화)
- \`CHANGELOG.md\` — Added (N-3/N-4 warnings) + Changed (N-1 splits, \`root\` clarification)
- \`ROADMAP.md\` — "Recently landed"에 Phase N 블록 추가
- \`Tests/.../subContainerSharedBindingsMultipleRemaps.swift\` (NEW 스냅샷)

## 검증 체크리스트

- [x] \`swift build\` 전 타겟 통과
- [x] \`swift test\` 359 tests green (N-1 직전 349 → +10: N-3 6개, N-4 4개)
- [x] \`swift test --filter InnoDIMacrosTests\` 153 macro tests green
- [x] 매크로 스냅샷 byte-identical (N-1 무영향, N-5 1개 신규 추가)
- [x] 공개/package API 표면 불변

## 코드 리뷰 반영

4개 sub-phase 모두 code-reviewer 에이전트로 리뷰한 뒤 should-fix 항목 반영:

- **N-1**: dead import 2개 제거 (\`SwiftSyntaxMacros\`, \`Foundation\`)
- **N-3**: 게이트 강화 (\`subsWithErrors\` set으로 다른 \`sub.*\` 에러가 있는 sub에 중복 경고 방지), 진단 앵커를 \`bindingSyntax.pattern\`으로 타이트닝, 진단 문구 수정, 3개 테스트 추가 (\`@SubContainer\`-only child / 제네릭 child / 도달 불가 child), 탐지 한계 문서화
- **N-4**: 비-제네릭 alias 테스트 추가, 진단 문구 50% 단축, 중첩 alias / 앵커 fallback / optional 비-언래핑 주석 추가
- **N-5**: "11 files" 오류 → "9 new siblings / 13 destination files" 정정 (MIGRATION+CHANGELOG+ROADMAP), MIGRATION \`SwiftUIMacros\` 행 voice 통일, 한국어 매트릭스 row 다듬기, 스냅샷 trailing newline, CHANGELOG 빈 줄 정리

## Notes — N-2 생략 이유

원 계획의 N-2 (\`Lazy<T>\` / \`Provider<T>\` conditional \`Sendable\` conformance)는 subsequent commits에서 이미 반대 방향으로 확정됨:

- \`Lazy<T>\` / \`Provider<T>\`를 non-\`Sendable\`로 의도적 유지 (\`InnoDI.swift\` doc comments + \`DIContainer.docc\`)
- \`StrictConcurrencyBuildTests.deferredWrappersStillFailInSendableHolders\` 테스트가 이 결정을 강제
- Sub-container 빌더 클로저는 이미 \`@Sendable\` 부착 완료 (\`subContainerBuildClosurePeerDecl\`)
- \`_LazyCell\`은 이미 \`NSLock\` + \`@unchecked Sendable\`

따라서 N-2의 작업 중 **의미 있는 것들은 이미 완료**되어 있고, **나머지는 반대 방향으로 확정**된 상태라 skip.

## Phase N 범위 밖 → 후속 과제

- Cross-module \`sub.child-overrides-missing\` 감지: \`ContainerSemanticBuildValidator\`에 child scope 플래그 (\`hasSharedMembers\`/\`hasTransientMembers\`/\`hasSubContainerMembers\`) plumbing 필요. 별도 PR로 분리 (plan의 N-3 build-support 부분).

## Base

이 PR은 #25 (\`codex/refactor-hierarchy-validator-prune-tca-example\`) 위에 stacked. #25 머지 후 \`git rebase origin/main\`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

# 릴리스 노트

* **New Features**
  * 그래프 렌더링에서 `root:` 파라미터가 엔트리 포인트 역할하여 루트 컨테이너에서 도달 가능한 서브그래프만 표시
  * 모든 `@DIContainer`에서 override 스캐폴딩 자동 생성 (사용자 정의 시 억제 가능)
  * `@SubContainer(bindings:)` 파라미터 추가로 자식 입력 레이블 재명명 지원

* **Bug Fixes**
  * `@DIContainer(validateDAG: false)` 동작 확장: 매크로 로컬 그래프 검증 진단 건너뛰기 (구조 진단은 유지)
  * 타입 별칭으로 도달하는 `Lazy<T>` / `Provider<T>` 감지하여 경고 추가

* **Documentation**
  * `root`와 `validateDAG` 파라미터 동작 명확화 및 조합 관계 문서화
  * 마이그레이션 가이드 및 변경 로그 업데이트

### CODE REVIEW

File: Sources/InnoDIMacros/DISubContainerChildOverridesCheck.swift [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+    // non-container, the existing `sub.*` diagnostics already cover the
+    // mismatch at other sites — don't double up here.
+    guard declGroupHasDIContainerAttribute(childDeclGroup) else {
+        return .unknown
+    }
+
+    if declGroupHasOverrideableMember(childDeclGroup) {
+        return .hasOverrideableMember
+    }
+    return .inputOnly(childContainerName: childTypeName)
```
Comment (Ethan-IS) [inline Sources/InnoDIMacros/DISubContainerChildOverridesCheck.swift]: This looks like a false positive now. `DIContainerCodeGenerator.generateAll` unconditionally synthesizes the overrides scaffolding for every `@DIContainer` (including input-only ones), and `SubContainerRuntimeTests.inputOnlyChildOverrideClosureCompilesAndExecutes` already asserts that `childOverrides` compiles and runs for an input-only child. Returning `.inputOnly` here means valid same-file `@SubContainer` uses now emit `sub.child-overrides-missing` warnings even though `<Child>.Overrides` does exist.

**Review by Ethan-IS — COMMENTED** [review]
One additional should-fix issue stood out on top of the existing README threads: the new `sub.child-overrides-missing` check appears to warn on input-only child containers even though input-only `@DIContainer`s now synthesize empty `Overrides` scaffolding and already have runtime coverage for `childOverrides` no-op forwarding. I left the inline note on the check itself.
